src/modules/launcher/Wox.Test/Plugins/WindowsIndexerTest.cs

```diff
@@ -125,40 +141,64 @@ public void ModifyQueryHelper_ShouldSetQueryHelper_WhenPatternDoesNotContainSplS
         }
 
         [Test]
-        public void ExecuteQuery_ShouldDisposeAllConnections_AfterFunctionCall()
```

Comment:
Is there a reason why this test was removed? It was there to ensure that the database and command reader connections are closed as expected. I think it should be added back.
This is a reference to an issue which was tracking the test cases to be added and this was one of the criteria (especially considering the fact that there are memory issues which we are addressing right now, it would nice to have it) - https://github.com/microsoft/Launcher/issues/5.

Reply:
@Alekhya.  @somil55 I personally tend to Unit test only the 'publically observable' side effect of a class. The one philosophical issue I have with this test is is seems to leak implementation details strictly for the purpose of unit testing.  In other words our unit tests couple the OldDbConnections to the WindowsIndexerApi.  I would propose the following.  Take this with a grain of salt (it's just my opnion) :). 

1.  Make the following members __private__. 
```
        public OleDbConnection conn;
        public OleDbCommand command;
        public OleDbDataReader WDSResults;
```
2.  Create a __protected__ method IsDisposed, that checks the connection state and also catches the ObjectDisposedException by executing the methods previously used on the command, connection, and WSDResults. 

3.  Leverage the [Private Object Class](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.privateobject?view=mstest-net-1.2.0) to assert that IsDisposed is true. 

Reasoning:   I think it's important to assert that we are returning the windows search api to the expected state after running a query, however I wouldn't want clients of this class to have to know specifics about the internal implementation details of how the query is made.   Adding a protected IsDisposed method should allow us to do that.  The only reason I suggested that this was that this be made protected is that it would have performance implications to validate that each of the internal objects have been disposed, and shouldnt be publicly exposed without a strong reason for it.

Reply:
Sounds good @ryanbodrug-microsoft . I think it would be a neat implementation without having to make those objects public just for the for the unit test. Thanks for the suggestion. Shall keep it in mind going forward as well :)

Reply:
Thanks for insights @ryanbodrug-microsoft. I think it is a neat solution. I will modify this PR to update this.

Reply:
I have added `HaveAllDisposableItemsBeenDisposed` method to check if all connections have been closed. The method couldn't be made protected because `Private Object Class` has been discontinued after MSTest 1.4.